### PR TITLE
Improve expiry options for the authentication token of the Content API in the subscriber config. [SDESK-5707]

### PR DESCRIPTION
**Pull request by GyanP**

### CODE REVIEW

File: scripts/apps/publish/views/subscribers.html [hunk 1]
```diff
@@ -265,12 +268,9 @@ <h3 class="modal__heading" translate ng-hide="subscriber._id">Add New Subscriber
...
                                             <select class="sd-line-input__select"
                                                 ng-model="tokens.ttl"
+                                                ng-disabled="tokens.neverExpire"
                                                 title="{{ :: 'Token expiry' | translate }}">
-                                                <option value="7" translate>1 week</option>
-                                                <option value="14" translate>2 weeks</option>
-                                                <option value="30" translate>1 month</option>
-                                                <option value="180" translate>6 months</option>
-                                                <option value="365" translate>1 year</option>
+                                                <option ng-repeat="(key, value) in tokens.days" value="{{value}}" translate>{{key}}</option>
```
Comment (pablopunk) [inline scripts/apps/publish/views/subscribers.html:273]: now that values are dynamic we lose the `translate` feature, don't we?

Comment (GyanP) [reply scripts/apps/publish/views/subscribers.html:273]: Yes, you are right.

**Review by pablopunk — APPROVED** [review]

File: scripts/apps/publish/controllers/SubscriberTokenController.ts [hunk 2]
```diff
@@ -52,8 +52,7 @@ export function SubscriberTokenController($scope, api, $rootScope) {
     this.generate = (ttl) =>
         api.save('subscriber_token', {
             subscriber: subscriber._id,
-            expiry: !this.neverExpire ? expiry(ttl) : null,
-            never_expire: this.neverExpire,
+            expiry_days: this.neverExpire ? null : ttl,
```
Comment (petrjasek) [inline scripts/apps/publish/controllers/SubscriberTokenController.ts:55]: I think this shouldn't be ever `null` but rather `0` for no expiry

File: scripts/apps/publish/views/subscribers.html [hunk 3]
```diff
@@ -236,8 +236,11 @@ <h3 class="modal__heading" translate ng-hide="subscriber._id">Add New Subscriber
...
                                     <div class="sd-list-item__column sd-list-item__column--grow sd-list-item__column--no-border">
                                         <div class="sd-list-item__row">
-                                            <span class="sd-overflow-ellipsis sd-list-item--element-grow"><code contenteditable="true" spellcheck="false">{{ :: token._id}}</code></span>
-                                            <span class="sd-list-item__text-label">{{ :: 'expires:' | translate }} <time sd-datetime data-date="token.expiry" data-from-now="true"></time></span>
+                                            <span class="sd-overflow-ellipsis sd-list-item--element-grow"><code contenteditable="false" spellcheck="false">{{ :: token._id}}</code></span>
+                                            <span class="sd-list-item__text-label">{{ :: 'expires:' | translate }}
+                                                <time ng-if="token.expiry" sd-datetime data-date="token.expiry" data-from-now="true"></time>
+                                                <span ng-if="token.never_expire" translate>Never Expire</span>
```
Comment (petrjasek) [inline scripts/apps/publish/views/subscribers.html:242]: I guess here it should check `ng-if="token.expiry == null"`

Comment (petrjasek) [reply scripts/apps/publish/views/subscribers.html:242]: and it might say just `Never` there is already `expires:` before

**Pull request merged**